controllers/neutronapi_controller.go

```diff
@@ -787,6 +787,29 @@ func (r *NeutronAPIReconciler) reconcileNormal(ctx context.Context, instance *ne
 	}
 	// create Deployment - end
 
+	sbCluster, err := ovnclient.GetDBClusterByType(ctx, helper, instance.Namespace, map[string]string{}, ovnclient.SBDBType)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	_, err = sbCluster.GetExternalEndpoint()
+	if err != nil {
+		r.Log.Info("No external endpoint defined for SB OVNDBCluster, deleting external Secrets")
+		cleanupSecretsErr := r.deleteExternalSecrets(ctx, helper, instance)
+		if cleanupSecretsErr != nil {
+			r.Log.Error(cleanupSecretsErr, "Failed to delete external Secrets")
+			return ctrl.Result{}, cleanupSecretsErr
+		}
+		return ctrl.Result{}, nil
+	} else {
+		// Create Secrets for external dataplane consumption
+		// TODO(ihar) - is there any hashing mechanism for EDP config? do we trigger deploy somehow?
```

Comment:
This is a good point. I suggested to @slagle before to:
1) keep the hash of the CM/Secret used for a given deployment of a OpenStackDataPlaneService recorded somewhere (potentially in the Status of the OpenStackDataPlaneService). This way the human operator can compare the hash in the neutron (nova) CR with the hash in the related OpenstackDataPlaneService CR to see if there is a need to redeployment as the config changed.
2) watch the CMs/Secrets the OpenstackDataPlaneService uses and either automatically or with a human acknowledgement do the redeployment of the service if the configuration changes. 

See the future enhancement section of the downstream doc https://docs.google.com/document/d/1tr5KzgT0cB-nCF36kSmEbzxy9i38oLQyfCsGwgQUiY8/edit#heading=h.fxf6yuwcf55m

Reply:
Ack, I will leave the todo entry for now.